xbmc/pvr/windows/GUIWindowPVRSearch.cpp

```diff
@@ -241,6 +240,14 @@ bool CGUIWindowPVRSearch::OnMessage(CGUIMessage &message)
       }
     }
   }
+  else if (message.GetMessage() == GUI_MSG_NOTIFY_ALL)
+  {
+    if (message.GetParam1() == GUI_MSG_UPDATE_SOURCES)
+    {
+      // removable drive connected/disconnected. base class triggers a window content refresh, which makes no sense for pvr search window.
+      return true;
+    }
+  }
```

Comment:
or, even more generic, move it to WindowBase and check m_allowNonLocalSources of the actual virtual directory?